### PR TITLE
Fix documentation about how X-Original-* headers are populated

### LINKED ISSUE #32235: Incorrect information for how X-Original headers are populated
### Description

The article says:

> When processed, X-Forwarded-{For|Proto|Host} values are moved to X-Original-{For|Proto|Host}

However, according to https://github.com/dotnet/aspnetcore/issues/54904#issuecomment-2033680580, that is not correct.

Rather than the `X-Original-*` headers storing the original values of the corresponding `X-Forwarded-*` headers, they instead store original information from the request before it was modified by the middleware.

See https://github.com/dotnet/aspnetcore/issues/54904#issuecomment-2033680580 for a detailed explanation for how the `X-Original-*` headers are populated.

### Page URL

https://learn.microsoft.com/en-us/aspnet/core/host-and-deploy/proxy-load-balancer?view=aspnetcore-8.0

### Content source URL

https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/host-and-deploy/proxy-load-balancer.md

### Document ID

c48abb08-bea3-dcff-c067-2edf5dc9b3a6

### Article author

@Rick-Anderson

### PR DESCRIPTION
**Pull request by joegoldman2**
Fixes #32235.
Fixes dotnet/aspnetcore#54904.

cc @Tratcher.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/host-and-deploy/proxy-load-balancer.md](https://github.com/dotnet/AspNetCore.Docs/blob/7648fbbeea76f4c0a6824862a794ee4e57b0fcb5/aspnetcore/host-and-deploy/proxy-load-balancer.md) | [Configure ASP.NET Core to work with proxy servers and load balancers](https://review.learn.microsoft.com/en-us/aspnet/core/host-and-deploy/proxy-load-balancer?branch=pr-en-us-32236) |

### CODE REVIEW

**Comment by guardrex** [comment]
@tdykstra @Rick-Anderson @wadepickett ... Everything was sailing along well this morning. [My last PR](https://github.com/dotnet/AspNetCore.Docs/pull/32234) built and merged fine, but a [workflow error](https://github.com/dotnet/AspNetCore.Docs/actions/runs/8539395660) came about and the *Merge to Live* PR wasn't automatically generated. This PR, too, is having trouble. There are no build report errors here, but the build is broken.

**Comment by guardrex** [comment]
BTW @joegoldman2 ... We haven't been code fencing these in this part of the doc, but we should because we don't want to take a chance on machine or human translation changing them ...

```diff
- HttpContext.Request.Host
+ `HttpContext.Request.Host`
```

Do you mind updating the PR to code-fence them?

**Comment by joegoldman2** [comment]
Fore sure! Something like:
```diff
- HttpContext.Request.Host
+ `[HttpContext.Request.Host](xref:Microsoft.AspNetCore.Http.HttpRequest.Host)`
```
?
Or should I remove the link?

**Comment by guardrex** [comment]
Put them both around the text of the link like this in each case ...

```
[`HttpContext.Request.Host`](xref:Microsoft.AspNetCore.Http.HttpRequest.Host)
```

**Comment by guardrex** [comment]
Excellent! Thanks for taking care of that.

Rick will be along in a bit. He manages this article, so he'll take the lead on the review.

WRT this not building properly, it might clear up on its own. Sometimes the build system flakes out for no obvious reason and then comes back up. We'll see in a sec with your last commit.

**Comment by guardrex** [comment]
All good 👍🍻 ... The build system got over it's hiccups.

File: aspnetcore/host-and-deploy/proxy-load-balancer.md [hunk 1]
```diff
@@ -37,9 +37,9 @@ The [Forwarded Headers Middleware](https://github.com/dotnet/aspnetcore/blob/mai
...
-* [HttpContext.Request.Scheme](xref:Microsoft.AspNetCore.Http.HttpRequest.Scheme): Set using the [`X-Forwarded-Proto`](https://developer.mozilla.org/docs/Web/HTTP/Headers/X-Forwarded-Proto) header value.
-* [HttpContext.Request.Host](xref:Microsoft.AspNetCore.Http.HttpRequest.Host): Set using the `X-Forwarded-Host` header value.
+* [`HttpContext.Connection.RemoteIpAddress`](xref:Microsoft.AspNetCore.Http.ConnectionInfo.RemoteIpAddress): Set using the `X-Forwarded-For` header value. Additional settings influence how the middleware sets `RemoteIpAddress`. For details, see the [Forwarded Headers Middleware options](#forwarded-headers-middleware-options). The consumed values are removed from `X-Forwarded-For`, and the old value of [`HttpContext.Connection.RemoteIpAddress`](xref:Microsoft.AspNetCore.Http.ConnectionInfo.RemoteIpAddress) is persisted in `X-Original-For`.
```
Comment (Tratcher) [inline aspnetcore/host-and-deploy/proxy-load-balancer.md:40]: Note this process may be repeated several times if there are multiple values in X-Forwarded-For/Proto/Host, resulting in several values moved to X-Original-*, including the original RemoteIpAddress/Host/Scheme.

Comment (Rick-Anderson) [reply aspnetcore/host-and-deploy/proxy-load-balancer.md:40]:
```suggestion
* [`HttpContext.Connection.RemoteIpAddress`](xref:Microsoft.AspNetCore.Http.ConnectionInfo.RemoteIpAddress): Set using the `X-Forwarded-For` header value. Additional settings influence how the middleware sets `RemoteIpAddress`. For details, see the [Forwarded Headers Middleware options](#forwarded-headers-middleware-options). The consumed values are removed from `X-Forwarded-For`, and the old value of [`HttpContext.Connection.RemoteIpAddress`](xref:Microsoft.AspNetCore.Http.ConnectionInfo.RemoteIpAddress) is persisted in `X-Original-For`.

Note: This process may be repeated several times if there are multiple values in `X-Forwarded-For/Proto/Host`, resulting in several values moved to `X-Original-*`, including the original `RemoteIpAddress/Host/Scheme`.
```

File: aspnetcore/host-and-deploy/proxy-load-balancer.md [hunk 2]
```diff
@@ -37,9 +37,11 @@ The [Forwarded Headers Middleware](https://github.com/dotnet/aspnetcore/blob/mai
...
+
+   Note: This process may be repeated several times if there are multiple values in `X-Forwarded-For/Proto/Host`, resulting in several values moved to `X-Original-*`, including the original `RemoteIpAddress/Host/Scheme`.
```
Comment (Rick-Anderson) [inline aspnetcore/host-and-deploy/proxy-load-balancer.md:42]:
```suggestion
* [`HttpContext.Connection.RemoteIpAddress`](xref:Microsoft.AspNetCore.Http.ConnectionInfo.RemoteIpAddress): Set using the `X-Forwarded-For` header value. Additional settings influence how the middleware sets `RemoteIpAddress`. For details, see the [Forwarded Headers Middleware options](#forwarded-headers-middleware-options). The consumed values are removed from `X-Forwarded-For`, and the old value of [`HttpContext.Connection.RemoteIpAddress`](xref:Microsoft.AspNetCore.Http.ConnectionInfo.RemoteIpAddress) is persisted in `X-Original-For`. Note: This process may be repeated several times if there are multiple values in `X-Forwarded-For/Proto/Host`, resulting in several values moved to `X-Original-*`, including the original `RemoteIpAddress/Host/Scheme`.
```

Comment (Rick-Anderson) [inline aspnetcore/host-and-deploy/proxy-load-balancer.md:42]:
```suggestion
* [`HttpContext.Connection.RemoteIpAddress`](xref:Microsoft.AspNetCore.Http.ConnectionInfo.RemoteIpAddress): Set using the `X-Forwarded-For` header value. Additional settings influence how the middleware sets `RemoteIpAddress`. For details, see the [Forwarded Headers Middleware options](#forwarded-headers-middleware-options). The consumed values are removed from `X-Forwarded-For`, and the old value of [`HttpContext.Connection.RemoteIpAddress`](xref:Microsoft.AspNetCore.Http.ConnectionInfo.RemoteIpAddress) is persisted in `X-Original-For`. Note: This process may be repeated several times if there are multiple values in `X-Forwarded-For/Proto/Host`, resulting in several values moved to `X-Original-*`, including the original `RemoteIpAddress/Host/Scheme`.
```

File: aspnetcore/host-and-deploy/proxy-load-balancer.md [hunk 3]
```diff
@@ -216,7 +216,7 @@ END of COMMENTED OUT -->
...
 
-When processed, `X-Forwarded-{For|Proto|Host}` values are moved to `X-Original-{For|Proto|Host}`. If there are multiple values in a given header, Forwarded Headers Middleware processes headers in reverse order from right to left. The default `ForwardLimit` is `1` (one), so only the rightmost value from the headers is processed unless the value of `ForwardLimit` is increased.
+If there are multiple values in a given header, Forwarded Headers Middleware processes headers in reverse order from right to left. The default `ForwardLimit` is `1` (one), so only the rightmost value from the headers is processed unless the value of `ForwardLimit` is increased.
 
 The request's original remote IP must match an entry in the <xref:Microsoft.AspNetCore.Builder.ForwardedHeadersOptions.KnownProxies> or <xref:Microsoft.AspNetCore.Builder.ForwardedHeadersOptions.KnownNetworks> lists before forwarded headers are processed. This limits header spoofing by not accepting forwarders from untrusted proxies. When an unknown proxy is detected, logging indicates the address of the proxy:
```
Comment (Rick-Anderson) [inline aspnetcore/host-and-deploy/proxy-load-balancer.md:221]:
```suggestion
The request's original remote `IP` must match an entry in the <xref:Microsoft.AspNetCore.Builder.ForwardedHeadersOptions.KnownProxies> or <xref:Microsoft.AspNetCore.Builder.ForwardedHeadersOptions.KnownNetworks> lists before forwarded headers are processed. This limits header spoofing by not accepting forwarders from untrusted proxies. When an unknown proxy is detected, logging indicates the address of the proxy:
```

**Pull request merged**